test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua

```diff
@@ -0,0 +1,131 @@
+local utils = require("user_modules/utils")
+local SDL = require("SDL")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local requestVrAddCommand = {
+  cmdID = 111,
+  type = "Choice",
+  vrCommands = { "Choice111" }
+}
+
+local requestChoiceParams = {
+  interactionChoiceSetID = 1,
+  choiceSet = {
+    {
+      choiceID = 111,
+      menuName = "Choice111",
+      vrCommands = { "Choice111" }
+    }
+  }
+}
+
+local deleteChoiceSetID = {
+  interactionChoiceSetID = requestChoiceParams.interactionChoiceSetID
+}
+
+local deleteResponseVrParams = {
+  cmdID = requestChoiceParams.choiceID,
+  type = "Choice"
+}
+
+local performParams = {
+  initialText = "TextInitial",
+  interactionMode = "MANUAL_ONLY",
+  interactionChoiceSetIDList = { requestChoiceParams.interactionChoiceSetID },
+  initialPrompt = {
+    { type = "TEXT", text = "pathToFile1" }
+  }
+}
+
+--[[ Local Functions ]]
+local function createInteractionChoiceSet(pChoiceParams)
+  local cid = common.mobile.getSession():SendRPC("CreateInteractionChoiceSet", pChoiceParams)
+
+  common.hmi.getConnection():ExpectRequest("VR.AddCommand", requestVrAddCommand)
+  :Do(function(_, data)
+      common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+
+  common.mobile.getSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+  common.mobile.getSession():ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+      hashId = data.payload.hashID
+    end)
+end
+
+local function sendPerformInteraction_REJECTED()
+  local cid = common.mobile.getSession():SendRPC("PerformInteraction", performParams)
+  common.mobile.getSession():ExpectResponse(cid, { success = false, resultCode = "REJECTED" })
+end
+
+local function deleteInteractionChoiceSet(pDeleteChoiceParams, pDeleteVrCommandParams)
+  local cid = common.mobile.getSession():SendRPC("DeleteInteractionChoiceSet", pDeleteChoiceParams)
+  common.hmi.getConnection():ExpectRequest("VR.DeleteCommand", pDeleteVrCommandParams)
+  :Do(function(_, data)
+      sendPerformInteraction_REJECTED()
+      local function deleteCommandResp()
+        common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+      end
+      common.run.runAfter(deleteCommandResp, 1000)
+    end)
+  common.mobile.getSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function reRegisterApp(pParams)
+  common.app.getParams().hashID = hashId
+  common.app.registerNoPTU()
+  common.getHMIConnection():ExpectRequest("VR.AddCommand", requestVrAddCommand)
+  :Do(function(_, data)
+      common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+end
+
+local function unexpectedDisconnect()
+  common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+  :Times(common.mobile.getAppsCount())
+  common.mobile.disconnect()
+  utils.wait(1000)
```

Comment:
`utils.wait` can be replaced by `common.run.wait`

Reply:
Done